### ISSUE TITLE
integrate DDC inference with new task model

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/dev_compiler#261]
we need to replace LibraryResolverWithInference with a new design that fits into the task model. It should be possible, somewhat similar to how constants are computed. This unblocks the new task model

**Comment by vsmenon** [comment]
@bwilkerson and @leafpetersen have more or less done this.  I'll close and open an new issue to track migrating ddc to the new task model code.
